tmt/steps/provision/testcloud.py

```diff
@@ -99,253 +114,268 @@
+        os.makedirs(TESTCLOUD_DATA, exist_ok=True)
+        os.makedirs(TESTCLOUD_IMAGES, exist_ok=True)
+
+        # Make sure libvirt domain template exists
+        GuestTestcloud._create_template()
+
+        import_testcloud()
 
         # Get configuration
-        config = testcloud.config.get_config()
+        self.config = testcloud.config.get_config()
 
-        # Make sure download progress is disabled, so it
-        # does not spoil our logging
-        config.DOWNLOAD_PROGRESS = False
+        # Make sure download progress is disabled unless in debug mode,
+        # so it does not spoil our logging
+        self.config.DOWNLOAD_PROGRESS = self.opt('debug') > 2
 
         # Configure to tmt's storage directories
-        config.DATA_DIR = TESTCLOUD_DATA
-        config.STORE_DIR = TESTCLOUD_IMAGES
+        self.config.DATA_DIR = TESTCLOUD_DATA
+        self.config.STORE_DIR = TESTCLOUD_IMAGES
 
-        # Initialize testcloud image
-        self.image = testcloud.image.Image(image_url)
+        # If image does not start with http/https/file, consider it a
+        # mapping value and try to guess the URL
+        if not re.match(r'^(?:https?|file)://.*', self.image_url):
+            self.image_url = GuestTestcloud._guess_image_url(self.image_url)
 
-        # Show which image we are using
-        self.info('image', f'{self.image.name}', 'green')
-
-        status = f'{self.image.name}'
+        # Initialize and prepare testcloud image
+        self.image = testcloud.image.Image(self.image_url)
+        self.verbose('qcow', self.image.name, 'green')
         if not os.path.exists(self.image.local_path):
-            self.info('status', 'downloading', 'green')
-
-        # prepare testcloud image
+            self.info('progress', 'downloading...', 'cyan')
         try:
             self.image.prepare()
         except FileNotFoundError:
-            raise GeneralError(
-                f"Could not find image '{self.image.local_path}'")
+            raise ProvisionError(f"Image '{self.image.local_path}' not found.")
 
+        # Create instance
+        self.instance_name = self._random_name()
         self.instance = testcloud.instance.Instance(
-            self.instance_name, image=self.image)
-
-        # generate ssh key
-        self.run(f'ssh-keygen -f {self.ssh_key} -N ""')
+            name=self.instance_name, image=self.image)
+        self.verbose('name', self.instance_name, 'green')
 
-        with open(self.ssh_pubkey, 'r') as pubkey:
-            config.USER_DATA = USER_DATA.format(
-                user_name=self.user, public_key=pubkey.read())
+        # Prepare ssh key
+        self.prepare_ssh_key()
 
-        self.info('status', 'booting', 'green')
-        self.instance.ram = self.option('memory') or DEFAULT_MEMORY
+        # Boot the virtual machine
+        self.info('progress', 'booting...', 'cyan')
+        self.instance.ram = self.memory
         self.instance.disk_size = DEFAULT_DISK_SIZE
```

Comment:
can we add an option also for this if we are on it pls?

Reply:
No problem, added.

Reply:
Both `memory` and `disk` now available:

```
  -m, --memory MEMORY  Set available memory in MB, 2048 MB by default.
  -D, --disk MEMORY    Specify disk size in GB, 10 GB by default.
```
Example output:
```
> tmt run provision -h virtual --memory 1024 --disk 12
/var/tmp/tmt/run-313

/plans/basic
    provision
        how: virtual
        image: fedora
        user: root
        memory: 1024 MB
        disk: 12 GB
```